components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -3855,4 +3880,62 @@ public static String getMountedPath(RegistryContext registryContext, String path
         return documentMap;
     }
 
+    /**
+     * this method used to set environments values to api object.
+     *
+     * @param environments environments values in json format
+     * @return set of environments that Published
+     */
+    public static Set<String> extractEnvironmentsForAPI(String environments) {
+        Set<String> environmentStringSet = null;
+        if (environments == null) {
+            environmentStringSet = new HashSet<String>(
+                    ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService()
+                                          .getAPIManagerConfiguration().getApiGatewayEnvironments().keySet());
+        } else {
+            //handle not to publish to any of the gateways
+            if ("none".equals(environments)) {
+                environmentStringSet = new HashSet<String>();
+            }
+            //handle to set published gateways nto api object
+            else if (!"".equals(environments)) {
```

Comment:
Can you please use isEmpty() method rather than "".equals() to check empty Strings?